Libraries/LibDNS/Message.cpp

```diff
@@ -1005,11 +1096,36 @@ ErrorOr<Records::DNSKEY> Records::DNSKEY::from_raw(ParseContext& ctx)
     // | ALGORITHM| an 8-bit value that identifies the public key's cryptographic algorithm.
     // | PUBLICKEY| the public key material.
 
+    u32 key_tag = 0;
     auto flags = static_cast<u16>(TRY(ctx.stream.read_value<NetworkOrdered<u16>>()));
+    key_tag += (bit_cast<u16>(NetworkOrdered<u16>(flags)) & 0xff) << 8;
+    key_tag += (bit_cast<u16>(NetworkOrdered<u16>(flags)) >> 8) & 0xff;
```

Comment:
Would a `bit_cast<NetworkOrdered<u16>>(flags) & 0xff << 8` work as well?
Not sure which is more idiomatic and or what is actually meant to happen here, especially as flags is read as network ordered and directly converted.
Note that Endian allows for implicit conversion

Reply:
that could actually just be `convert_between_host_and_blahblah`, should do that.